tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/HotChocolateTests.cs

```diff
@@ -0,0 +1,261 @@
+// </copyright>
+#if NET5_0_OR_GREATER || NETCOREAPP3_1_OR_GREATER
+
+#pragma warning disable SA1402 // File may only contain a single type
+#pragma warning disable SA1649 // File name should match first type name
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public class HotChocolate12Tests : HotChocolateTests
+    {
+        public HotChocolate12Tests(ITestOutputHelper output)
+            : base("HotChocolate", output, nameof(HotChocolate12Tests))
+        {
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(PackageVersions.HotChocolate), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        [Trait("SupportsInstrumentationVerification", "True")]
+        public async Task SubmitsTraces(string packageVersion)
+            => await RunSubmitsTraces(packageVersion);
+    }
+
+    [UsesVerify]
+    public abstract class HotChocolateTests : TestHelper
+    {
+        private const string ServiceVersion = "1.0.0";
+
+        private readonly string _testName;
+
+        protected HotChocolateTests(string sampleAppName, ITestOutputHelper output, string testName)
+            : base(sampleAppName, output)
+        {
+            SetServiceVersion(ServiceVersion);
+
+            _testName = testName;
+        }
+
+        protected async Task RunSubmitsTraces(string packageVersion = "")
+        {
+            SetInstrumentationVerification();
+            using var telemetry = this.ConfigureTelemetry();
+            int? aspNetCorePort = null;
+
+            using (var agent = EnvironmentHelper.GetMockAgent())
+            using (Process process = StartSample(agent, arguments: null, packageVersion: packageVersion, aspNetCorePort: 0))
+            {
+                var wh = new EventWaitHandle(false, EventResetMode.AutoReset);
+
+                process.OutputDataReceived += (sender, args) =>
+                {
+                    if (args.Data != null)
+                    {
+                        if (args.Data.Contains("Now listening on:"))
+                        {
+                            var splitIndex = args.Data.LastIndexOf(':');
+                            aspNetCorePort = int.Parse(args.Data.Substring(splitIndex + 1));
+
+                            wh.Set();
+                        }
+                        else if (args.Data.Contains("Unable to start Kestrel"))
+                        {
+                            wh.Set();
+                        }
+
+                        Output.WriteLine($"[webserver][stdout] {args.Data}");
```

Comment:
Just a note: As you've seen, sending the requests from the test is quite hard work, and has occasional issues with ports etc.

I prefer to have the application send requests to _itself_, similar to the way the GRPC apps are written. It simplifies a lot of this 

No need to change it now, we have plenty of tests that work like this, just an FYI 🙂

Reply:
I much prefer this, as it's generally a lot easier. I don't think it makes sense to try and _change_ the samples at this stage, just something to bear in mind for new samples.

> that somehow ties the tests to the samples code though

Well, they're kind of tied to the sample code no matter what you do 🤷‍♂️ I don't think that's an issue personally 🙂